File: 백준/Silver/3005. 크로스워드 퍼즐 쳐다보기/크로스워드 퍼즐 쳐다보기.js
const fs = require('fs');
// let input = fs.readFileSync('./input.txt').toString().trim().split('\n');

/**정답 제출 시에는 이 코드로 제출 */
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n,m] = input.shift().split(' ').map(Number);
let board = input.map((it) => it.trim().split(''))
let result = []

// 행일때
for (let i=0; i<n; i++) {
  let word = '';
  for (let j=0; j<m; j++) {
    if (board[i][j] !== '#') {
      word += board[i][j];
    } else {
      if (word.length > 1) {
        result.push(word);
        word = '';
      } else {
        word = '';
      }
    }
  }
  if (word.length > 1) {
    result.push(word)
  }
}

// js에서 2차원 배열 전환하는 법 
zip = rows => rows[0].map((_,c) => rows.map(row=>row[c]))
let z_board = zip([...board]);

// 열일때
for (let i=0; i<m; i++) {
  let z_word = '';
  for (let j=0; j<n; j++) {
    if (z_board[i][j] !== '#') {
      z_word += z_board[i][j];
    } else {
      if (z_word.length > 1) {
        result.push(z_word);
        z_word = '';
      } else {
        z_word = '';
      }
    }
  }
  if (z_word.length > 1) {
    result.push(z_word);
  }
}

result.sort();
console.log(result[0]);

File: 백준/Silver/3005. 크로스워드 퍼즐 쳐다보기/크로스워드 퍼즐 쳐다보기.py
import sys
input = sys.stdin.readline

R,C = map(int,input().split())
board = [list(input().rstrip()) for _ in range(R)]
tmp = []

# 행일때
for i in range(R):
    word = ''
    for j in range(C):
        if board[i][j] != '#':
            word += board[i][j]
        else:
            if len(word) >= 2:
                tmp.append(word)
                word = ''
            else:
                word = ''
                continue
    if len(word) > 1:
        tmp.append(word)

# 열일때
z_board = list(map(list,zip(*board)))
for i in range(C):
    word = ''
    for j in range(R):
        if z_board[i][j] != '#':
            word += z_board[i][j]
        else:
            if len(word) >= 2:
                tmp.append(word)
                word = ''
            else:
                word = ''
                continue
    if len(word) > 1:
        tmp.append(word)

tmp.sort()
print(tmp[0])